### ISSUE TITLE
A BinanceAPI error has occurred. Code = -2010

### ISSUE THREAD

**Issue by crunkpinoy** [fj317/PumpBot#88]
Hi,
I used to make profits of up to 90% on pumps doing it manually but since using this bot I'm always getting errors. I don't get any error on tests trying out different pairs and configs. Only on the actual pumps it doesn't work. I'm pretty fast buying the pumped coin manually but I was hoping to be even faster using the bot. Missing out on all pumps during the last 2 months trying different configs. Also using the latest version always hoping the issue was fixed.

config:
    "quotedCoin": "BTC",
    "buyLimit": 1,
    "percentOfWallet": 80,
    "manualQuoted": 0.00000000,
    "profitMargin": 180,
    "stopLoss": 0.90,
    "endpoint": "default",

cmd:
Caching all BTC pairs average prices...
This can take a while. Please, be patient...

Sucessfully cached BTC pairs!
Checking BNB balance if topup required.

 ___                                ___           _
(  _`\                             (  _`\        ( )_
| |_) ) _   _   ___ ___   _ _      | (_) )   _   | ,_)
| ,__/'( ) ( )/' _ ` _ `\( '_`\    |  _ <' /'_`\ | |
| |    | (_) || ( ) ( ) || (_) )   | (_) )( (_) )| |_
(_)    `\___/'(_) (_) (_)| ,__/'   (____/'`\___/'`\__)
                         | |
                         (_)

Investing amount for BTC: 0.0018328
Investing amount in USD: 109.08

Coin pair: VIA
A BinanceAPI error has occurred. Code = -2010
Account has insufficient balance for requested action.. Please use https://github.com/binance/binance-spot-api-docs/blob/master/errors.md to find greater details on error codes before raising an issue.
Quitting, this could take a few seconds!

Have enough BNB to cover the fees. Using just 80% of my BTC to avoid conflicts with price changes. Tried limit_buy with different values and market_buy as well. How can I have insufficient balance?
I know this error has been discussed before but I don't understand if it is a rounding issue or something else.
Also my log file remains empty. Running cmd as administrator won't let me run the bot.
Had other errors before but I guess the percentOfWallet 98 was too high and the buyLimit 1.10 too low. Now I always get insufficient balance error but as I said only during actual pumps. Non-pump is working fine.
Hope we can find a solution for this.

Thanks
crunkpinoy

**Comment by artia123-py** [comment]
Same for me. Before the pump bot works correctly. During the pump always same error when trying to buy coin

Coin pair: VIA
A BinanceAPI error has occurred. Code = -2010
Account has insufficient balance for requested action.. Please use https://github.com/binance/binance-spot-api-docs/blob/master/errors.md to find greater details on error codes before raising an issue.
Quitting, this could take a few seconds!

**Comment by kuratkull** [comment]
There has been a problem with the percentage value in calculations for a long time. I recommend setting it to 50%
( Also, let me give you some information you need to pay attention to yesterday's pump. The number of people doing VIA BTC PUMP at 20:03:15 when the pump started is 7 and the price reached 0.0007500 The number of people doing VIA BTC PUMP at 20:04:25 when the pump started is 3 and the price reached 0.0010200

first they buy and sell them. and you take it from high.
You do not earn your money easily, please be careful.
sorry i didn't take a screenshot, you could understand what i wanted to say.

**Comment by kuratkull** [comment]
Let me give an example pump image

2021-04-13 02:30:30.440000  |  ZECBTC           |  Ticks:3 |  %:0.35    |  %:0.35   | Price: 0.003735
2021-04-13 02:30:29.960000  |  ZRXBTC           |  Ticks:1 |  %:-0.14   |  %:0.14   | Price: 3.54e-05
2021-04-13 02:30:30.336000  |  ALPHAUSDT    |  Ticks:1 |  %:-0.17   |  %:0.17   | Price: 1.626
2021-04-13 02:30:31.418000  |  ZECBTC           |  Ticks:4 |  %:0.37    |  %:0.37   | Price: 0.003736
2021-04-13 02:30:32.366000  |  RAMPBUSD         |  Ticks:2 |  %:0.34    |  %:0.49   | Price: 0.8229
2021-04-13 02:30:32.409000  |  NEOUSDC          |  Ticks:1 |  %:-0.39   |  %:0.39   | Price: 64.225
2021-04-13 02:30:33.427000  |  ALPHAUSDT        |  Ticks:2 |  %:-0.03   |  %:0.30   | Price: 1.62991
2021-04-13 02:30:33.233000  |  QSPBTC           |  Ticks:1 |  %:-0.55   |  %:0.55   | Price: 1.82e-06

2021-04-13 02:30:37.385000  |  CHRBTC           |  Ticks:2 |  %:0.00    |  %:0.33   | Price: 6.15e-06

2021-04-13 02:30:39.407000  |  NEBLBTC          |  Ticks:2 |  %:-0.86   |  %:0.86   | Price: 7.664e-05
2021-04-13 02:30:38.906000  |  HOTEUR           |  Ticks:2 |  %:0.25    |  %:0.63   | Price: 0.01593
2021-04-13 02:30:42.423000  |  ZENBNB           |  Ticks:1 |  %:0.87    |  %:0.87   | Price: 0.127
2021-04-13 02:30:42.430000  |  ORNUSDT          |  Ticks:2 |  %:-0.67   |  %:0.67   | Price: 14.7885
2021-04-13 02:30:43.446000  |  CHRBTC           |  Ticks:3 |  %:-0.33   |  %:0.65   | Price: 6.12e-06
2021-04-13 02:30:43.156000  |  CHZGBP           |  Ticks:3 |  %:0.60    |  %:0.60   | Price: 0.4166
2021-04-13 02:30:43.437000  |  LINKDOWNUSDT             |  Ticks:3 |  %:0.21    |  %:0.41   | Price: 5.041751
2021-04-13 02:30:43.044000  |  CELRBTC          |  Ticks:1 |  %:0.97    |  %:0.97   | Price: 1.04e-06

**Comment by kuratkull** [comment]
@fj317 Do you have a chance to fix it as **_quentity = balance / coinprice_**, not as a percentage? then we can make the maximum amount of purchases binance is having trouble with percentage calculation here...

**Comment by fj317** [comment]
> 
> 
> @fj317 Do you have a chance to fix it as **_quentity = balance / coinprice_**, not as a percentage? then we can make the maximum amount of purchases binance is having trouble with percentage calculation here...

I'm confused by what you mean here? 
The code that works out the quantity is:
```
amountOfCoin = AmountToSell / float(averagePrice)
```
Where AmountToSell is the whatever the user's balance mutiplied by their percentOfWallet parameter; and averagePrice is the cached average price of the coinPair.

**Comment by crunkpinoy** [comment]
Thanks for the reply. 50% is actually the default config. I will try that next time. However the buylimit of 1.15 is too low. It is pumping too fast and is far above 15% until my order is placed. Obviously the coins are always pre-pumped. The VIA pump was actually suspected by some groups to be the pump coin and many bought it already days before the pump. Also some of the previous pumps were exposed. Seems the host is feeding the coins with obviously huge amounts which makes it easier to detect.
Just saw you gave it a bug label. Until it is fixed I will try the 50% size.
Thanks

**Comment by artia123-py** [comment]
crunkpinoy can you please share the groups which shared info about VIA pump ?

**Comment by crunkpinoy** [comment]
Yes artia I can't remember because I'm in many groups on telegram and searching for VIA gives too many results. You can just search for "binance signals" but be careful because there are also so many scammers. Besides of pumps some also provide good signals for scalping, swing trades etc. but I recommend to observe those signals first to know which channels are actually giving profitable information.

**Comment by RohitKonge** [comment]
@crunkpinoy Does the Pump bot work now? We can make a 'market order' to quickly buy the coins.

**Comment by RohitKonge** [comment]
@fj317 Does the Pump bot work now?

**Comment by RohitKonge** [comment]
@fj317 Also will it work if we set percentOfWallet = 100 ? 
Or will there be an error ?
